deployer/src/deployer/analyze_pr.py

```diff
@@ -0,0 +1,191 @@
+import json
+from collections import defaultdict
+from pathlib import Path
+
+from github import Github
+from selectolax.parser import HTMLParser
+
+from .utils import log
+
+
+def analyze_pr(build_directory: Path, config):
+    """Given a directory of documents built from a PR, look through it and
+    post a GitHub PR comment or just print it to stdout."""
+
+    combined_comments = []
+
+    if config["prefix"]:
+        combined_comments.append(post_about_deployment(build_directory, **config))
+
+    if config["analyze_flaws"]:
+        combined_comments.append(post_about_flaws(build_directory, **config))
+
+    if config["analyze_dangerous_content"]:
+        combined_comments.append(
+            post_about_dangerous_content(build_directory, **config)
+        )
+
+    combined_comment = "\n\n".join(x for x in combined_comments if x)
+
+    if not combined_comment:
+        print("Warning! Nothing to comment at all!")
+        return
+
+    print("POST".center(80, "_"))
+    print()
+    print(combined_comment)
+    print()
+    print("END POST".center(80, "_"))
+
+    if not config["repo"]:
+        print("Warning! No 'repo' config")
+    elif not config["pr_number"]:
+        print("Warning! No 'pr_number' config")
+    elif config["repo"] and config["pr_number"]:
+        pr_url = f"https://github.com/{config['repo']}/pull/{config['pr_number']}"
+        if config["dry_run"]:
+            log.warning(f"Dry-run! Not actually posting any comment to {pr_url}")
+        else:
+            print(f"Posting to {pr_url}")
+
+            if config["github_token"]:
+                github = Github(config["github_token"])
+                github_repo = github.get_repo(config["repo"])
+                github_issue = github_repo.get_issue(number=int(config["pr_number"]))
+                github_issue.create_comment(combined_comment)
```

Comment:
This is cool! I love that there's a Python package for doing this!

Reply:
I messed around for ages with pyoctokit and it was just a painful mess.